### ISSUE TITLE
Excessive logging for s3: Failed to find credentials process's profile

### ISSUE THREAD

**Issue by anotherbugmaster** [ClickHouse/ClickHouse#30559]
When selecting from S3 table with `use_environment_credentials=true` CH writes lots of messages like this

```
<Error> AWSClient: ProcessCredentialsProvider: Failed to find credential process's profile: default
```

I've dug in the CH and AWS SDK codebase a little, and it seems like AWS Client is supposed to cache credentials obtained using instance metastore in local filesystem and then use them till the expiration date comes, but that doesn't happen for some reason, hence this error.

I've also tried to save credentials obtained from metastore in `/nonexistent/.aws/` (clickhouse user home directory) manually, and these messages were gone before CH made another request to metastore and made credentials saved in a file obsolete.

**Does it reproduce on recent release?**

Yes, tested with 21.10.2.15

**How to reproduce**

* Which ClickHouse server version to use
	* 21.10.2.15
* Which interface to use, if matters
	* Doesn't matter
* Non-default settings, if any
	```
	<use_environment_credentials>true</use_environment_credentials>
	```
* `CREATE TABLE` statements for all tables involved
	```
		CREATE TABLE s3_table (
  			json String
		)
		ENGINE = S3(
  			'https://s3.amazonaws.com/bucket/prefix/*.gz',
  			'JSONAsString'
		);
	```
* Queries to run that lead to unexpected result
	```
	SELECT * FROM s3_table LIMIT 1000000;
	``` 

**Expected behavior**

Not to have lots of error messages in logs

**Error message and/or stacktrace**

```
<Error> AWSClient: ProcessCredentialsProvider: Failed to find credential process's profile: default
```

**Comment by krdenison** [comment]
Seeing the same behaviour in my companies deployment. We setup two S3 connections in our storage config and authenticate using a kubernetes ServiceAccount linking to an IAM role with permissions to the S3 bucket. 

Seeing tens of thousands of these messages in our error logs
```
$ head clickhouse-server.err.log -n 1 && grep "Failed to find credential process" -c clickhouse-server.err.log && tail clickhouse-server.err.log -n 1
2022.01.07 16:43:00.443533 [ 102 ] {} <Error> AWSClient: ProcessCredentialsProvider: Failed to find credential process's profile: default
37156
2022.01.07 18:20:14.715636 [ 79 ] {} <Error> AWSClient: ProcessCredentialsProvider: Failed to find credential process's profile: default
```

**Comment by filimonov** [comment]
Looks related to https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html 

https://github.com/aws/aws-sdk-cpp/blob/bb1fdce01cc7e8ae2fe7162f24c8836e9d3ab0a2/aws-cpp-sdk-core/source/auth/AWSCredentialsProvider.cpp#L380-L390

**Comment by anotherbugmaster** [comment]
Guys, this is serious. Excessive logging from this stuff has eaten 2 GB of disk space and 200% constant CPU usage. Does anyone has any more ideas about what could've caused it?

**Comment by alexey-milovidov** [comment]
Confirmed.

**Comment by krdenison** [comment]
Can this be back ported to the next 22.3 LTS release?

**Comment by lichenglin** [comment]
Hello, is there any way to fix it without upgrade my version is 21.12.3.32

**Comment by Soooooda** [comment]
Hi team, my version is 22.2.2.1, this problem still exists.
